### PR TITLE
change error message

### PR DESCRIPTION
**Pull request by TyOverby**
I didn't like the old error message

### CODE REVIEW

**Comment by TyOverby** [comment]
@dtolnay

**Comment by dtolnay** [comment]
Seems reasonable. Another option would be to add a SerializeError variant for this case.

**Comment by TyOverby** [comment]
I like that more actually.

**Comment by dtolnay** [comment]
I don't think "infinite" is the right word, especially because SizeLimit::Infinite means something completely different. If serialize_seq does not receive a size, it is because the Serialize type does not know the length up front, for example because we are serializing an iterator that can only be iterated once.

File: src/serde/writer.rs [hunk 1]
```diff
@@ -49,6 +49,7 @@ impl fmt::Display for SerializeError {
         match *self {
             SerializeError::IoError(ref err) => write!(f, "IoError: {}", err),
             SerializeError::Custom(ref s) => write!(f, "Custom Error {}", s),
+            SerializeError::InfiniteSequence => write!(f, "InfiniteSequence"),
```
Comment (dtolnay) [inline src/serde/writer.rs]: Please make this a human-readable error message. This is the Display impl not Debug.

**Comment by TyOverby** [comment]
@dtolnay Changed it to `SequenceMustHaveLength` and the Display impl mentions maps and seqs

**Comment by dtolnay** [comment]
Cool. I filed https://github.com/TyOverby/bincode/issues/98 to follow up on the other messages.

**Pull request merged**